src/rules/noStaticOnlyClassesRule.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+import { getChildOfKind,
+        hasModifier,
+        isClassDeclaration,
+        isConstructorDeclaration,
+        isParameterProperty } from "tsutils";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "no-static-only-classes",
+        description: Lint.Utils.dedent`
+            Disallows classes containing only static members. Classes
+            with non-empty constructors are ignored.`,
+        rationale: Lint.Utils.dedent`
+            Users who come from a Java-style OO language may wrap
+            their utility functions in an extra class, instead of
+            putting them at the top level.`,
+        optionsDescription: "Not configurable.",
+        options: null,
+        optionExamples: [true],
+        type: "functionality",
+        typescriptOnly: false,
+    };
+
+    public static FAILURE_STRING = "Classes containing only static members are disallowed.";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NoStaticOnlyClassesWalker(sourceFile, this.ruleName, this.ruleArguments));
+    }
+}
+
+class NoStaticOnlyClassesWalker extends Lint.AbstractWalker<string[]> {
+    public walk(sourceFile: ts.SourceFile) {
+        const checkIfStaticOnlyClass = (node: ts.Node): void => {
+            if (isClassDeclaration(node) && !isEmptyClass(node) && !hasExtendsClause(node)) {
+                for (const member of node.members) {
+                    if (isConstructorWithShorthandProps(member) ||
+                       (!isConstructorDeclaration(member) && !hasModifier(member.modifiers, ts.SyntaxKind.StaticKeyword))) {
+                        return;
```

Comment:
> I don't see much point in a class with static members and constructors only... Is there a real-life coding scenario where this is useful?

It can be useful for writing compilers, where classes simulate discriminated unions (much like "case classes" in Scala). For example:

```typescript
class While extends Statement {
    constructor(public condition: Expression, public body: Statement[]) {}
}
class Break extends Statement {}
class Continue extends Statement {}
```

Then you'd check if a statement is a `break` using `stmt instanceof Break`.